From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Thu, 20 Dec 2018 14:32:43 +0300
Subject: [PATCH] cifs: Use kzfree() to free password
Git-commit: 34bca9bbe7a81c4b53cacb20e996090ddf371c51
Patch-mainline: v5.0-rc1
References: bsc#1144333

We should zero out the password before we free it.

Fixes: 3d6cacbb5310 ("cifs: Add DFS cache routines")
Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Reviewed-by: Paulo Alcantara <palcantara@suse.de>
Acked-by: Aurelien Aptel <aaptel@suse.com>
---
 fs/cifs/dfs_cache.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/cifs/dfs_cache.c b/fs/cifs/dfs_cache.c
index f4e09073bb7d..cd63c4a70875 100644
--- a/fs/cifs/dfs_cache.c
+++ b/fs/cifs/dfs_cache.c
@@ -1127,7 +1127,7 @@ static int dup_vol(struct smb_vol *vol, struct smb_vol *new)
 err_free_unc:
 	kfree(new->UNC);
 err_free_password:
-	kfree(new->password);
+	kzfree(new->password);
 err_free_username:
 	kfree(new->username);
 	kfree(new);
-- 
2.16.4

